homeassistant/components/emulated_hue/upnp.py

```diff
@@ -160,3 +136,38 @@ def _prepare_response(self, search_target, unique_service_name):
 
 """
         return response.replace("\n", "\r\n").encode("utf-8")
+
+
+async def async_create_upnp_datagram_endpoint(
+    host_ip_addr: str,
+    upnp_bind_multicast: bool,
+    advertise_ip: str,
+    advertise_port: int,
+) -> UPNPResponderProtocol:
```

Comment:
I moved this so I could type the above, and made it a coro since we want to move this into an external lib in next steps

Reply:
I also want to get rid of all the manual config an use the network integration so there is less to rip out later